src/renate/utils/misc.py

```diff
@@ -0,0 +1,29 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Union
+from pathlib import Path
+
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
+
+
+def int_or_str(x: str) -> Union[str, int]:
```

Comment:
not sure we have a good location for this function but please check for type of x instead of try expect casting